lab2.js

```diff
@@ -85,28 +114,62 @@ var hello = {
 // speak, and method (that you'll place on the prototype) called
 // sayHello.
 
-function SentientBeing () {
+function SentientBeing(homeplanet, tounge) {
   // TODO: specify a home planet and a language
   // you'll need to add parameters to this constructor
+  this.homeplanet = homeplanet;
+  this.tounge = tounge;
+}
+
+function Klingon() {
+  this.species = 'Klingon';
 }
+Klingon.prototype = new SentientBeing('Qo\'nos', 'klingon');
+function Human() {}
+Human.prototype = new SentientBeing('Earth', 'federation standard');
+function Romulan() {}
+Romulan.prototype = new SentientBeing('Romulus', 'romulan');
 
 // sb is a SentientBeing object
-function sayHello (sb) {
+SentientBeing.prototype.sayHello = function(sb) {
     // TODO: say hello prints out (console.log's) hello in the
     // language of the speaker, but returns it in the language
     // of the listener (the sb parameter above).
     // use the 'hello' object at the beginning of this exercise
     // to do the translating
-
+    if (sb.tounge === 'klingon') {
+      return (hello.klingon);
+    } else if (sb.tounge === 'romulan') {
+      return (hello.romulan);
+    } else {
+      return (hello.federationstandard);
+    }
+    if (this.tounge === 'klingon') {
+      console.log(hello.klingon);
+    } else if (this.tounge === 'romulan') {
+      console.log(hello.romulan);
+    } else {
+      console.log(hello.federationstandard);
+    }
```

Comment:
this is a good use case for bracket notation. If the languages on the sentient being match what's being used for parameters in the hello variable, you can do:
console.log(hello[this.language]);
return hello[sb.language];